### PR TITLE
Update to Linux 5.11 drivers

### PR DESCRIPTION
**Pull request by dumbbell**
This branch synchronizes `drm-kmod` with Linux 5.11.

See https://github.com/freebsd/drm-kmod/pull/163#issuecomment-1157699164 for instructions to play with this branch.

**Status**:

### CODE REVIEW

**Comment by dumbbell** [comment]
I slightly lied on the progress :-) I cherry-picked only 484 out of 1334 patches (36%). I'm using this branch as of now to see how it behaves on a day-to-day basis.

**Comment by dumbbell** [comment]
The backport is completed for a few weeks now. I'm using this on a daily basis with an Intel CannonLake GPU (i915).

And starting from today, it seems to work with an discrete AMD Radeon RX 6700 XT! On top of that, the dGPU is plugged to my laptop using a Razer Core X Chroma Thunderbolt 3 external case :-) I will start to use that dGPU everyday to further test it.

Here is a screenshot using Sway/Wayland and this 6700 XT:

freebsd-rx-6700-xt

It depends on a FreeBSD src branch I didn't push yet.

**Comment by evadot** [comment]
Awesome news.
Any suspend/resume regression ? (either intel or amdgpu works perfectly here on multiple machines)
Also does amdgpu can still be unloaded and re-loaded (works here), i915kms still have some problems sometimes that I haven't been able to find and fix.

**Comment by nevillehay** [comment]
Thanks for all your hard work! I have an RX 6500XT. I'd like to test the 5.11 version. 
How can I do this?

**Comment by dumbbell** [comment]
I just updated the comment above with the freshly uploaded firmwares update.

**Comment by nevillehay** [comment]
Thanks a lot! I'll try this tomorrow or Monday. I take it this won't work on 13 STABLE?

**Comment by dumbbell** [comment]
I don't know. You could try to rebase the `freebsd-src`'s `linuxkpi-changes-for-drm-from-5.11` branch on top of `stable/13` and see if it compiles already.

**Comment by nevillehay** [comment]
Okay. Thanks.
I have to wait for a new CPU. Should be coming in the next couple of days.
Sorry for the delay. I also have an r9-270x. Should I use my RX 6500xt or
the r9-270x? The 270x works fine with drm-510-kmod.

**Comment by nevillehay** [comment]
On second thought I'll do everything with the 6500xt installed

**Comment by dumbbell** [comment]
Testing the r9-270x would also be a nice baseline test to spot any regressions. But any testing is welcome! So feel free to try whatever is more convenient of course!

**Comment by nevillehay** [comment]
Okay. Thanks. Do I need to install anything for compiling or will FreeBSD
14 with xfce desktop with src be enough. Do I need debugging tools?

**Comment by dumbbell** [comment]
Do you plan to install a new FreeBSD on a new partition or ZFS filesystem? Do you use boot environments?

If you are currently using FreeBSD 13.x, I would recommend to keep this one untouched, in case you hit any issues with 14-CURRENT that would block you. If you do a fresh install or duplicate your 13.x install into a new boot environment, you should be fine either way. It's been a very long time since I used the FreeBSD installer, but you should already have the base compiler available and it will be enough to recompile FreeBSD and the drm+firmwares kernel modules.

If you duplicate your existing install of 13.x, you can use that clone to do the testing with your existing ports and configuration. In my instructions, I only covered the compilation of the kernel, but that's ok: a 14-CURRENT kernel will work with a 13.x userland. As I said earlier, just make sure to not kldload the DRM drivers during boot, just to make sure to not break boot. Load the module manually at first.

**Comment by nevillehay** [comment]
Thanks for your quick replies. I'll do a fresh install of FreeBSD 14. I won't have FreeBSD 13 running at all. I have another computer running FreeBSD (actually GhostBSD) 13 stable so if the FreeBSD 14 goes belly up, it's not a problem

**Comment by dumbbell** [comment]
As for debugging configuration and tools, you need to enable kernel core dumps. I have `dump_dev="AUTO"` in `/etc/rc.conf`, I think it's enough. You may need to install gdb to get nice stacktraces as well. You can force a kernel crash with `sudo sysctl debug.kdb.panic=1` to test that code dumps are saved correctly. During the next reboot, the kernel core dump will be written to `/var/crash`.

If you get a kernel core dump while testing the driver, I will be interested in the latest `/var/crash/core.txt.$N` file (by default, they are written in round-robin from 0 to 9). That said, depending on when/where the kernel crashes in the video driver, the kernel may just hang (and a long press on the power button is required) or reboot without writing a core dump.

**Comment by nevillehay** [comment]
Okay thanks very much for the information. I ordered a Ryzen 7 5800x and a
cooler. They're coming tomorrow. When I'm up and running I'll let you know.

**Comment by nevillehay** [comment]
I got the new CPU, and got everything working fine on FreeBSD 13 stable. I
just installed FreeBSD 14 from June 17 and there seems to be some sort of
regression. Clean install with xfce desktop. I installed drm-kmod and this
pulled in drm-510-kmod I also install gpu -firmware-kmod but during the
boot process when it says loading amdgpu the screen goes black and stays
black. I also put in rc.conf

kld_list="amdgpu"

I'll have to try a different version.

Currently using scfb to get to the desktop

**Comment by nevillehay** [comment]
No I didn't. I noticed your instructions for that but how do you link
/usr/src to the clone? What is the command?
Steps 1 and 3 went great!
Again thanks for all your work.

**Comment by evadot** [comment]
> Did you link `/usr/src` to the freebsd-src clone? I think `/usr/src` contains or links to the vanilla FreeBSD sources.
> 
> There is a variable to indicate where to find the source tree but I don't recall it's name

export SYSDIR=/path/to/src/sys (the sys is important)

**Comment by nevillehay** [comment]
Thanks. Will do it tonight. Do I run this command first, then the others?

**Comment by dumbbell** [comment]
Thanks @evadot!

@nevillehay: Yes, you need to run this command before you try to compile drm-kmod. Note that if you change terminal or reboot, you'll have to run it again as the exported variable's scope is limited to a running shell instance.

When I suggesting to link, I had a symlink in mind, so something like `ln -s /path/to/src /usr/src` (assuming `/usr/src` does not exist). But you don't need both, the exported variable will work just fine.

**Comment by nevillehay** [comment]
By the way, there's small error here


   1.

   You need to checkout the FreeBSD src branch I mentionned,
   linuxkpi-changes-for-drm-from-5.11
   <https://github.com/dumbbell/freebsd-src/tree/linuxkpi-changes-for-drm-from-5.11>,
   and compile a kernel from that branch:

   git clone -b linuxkpi-changes-for-drm-from-5.11
https://github.com/dumbbell/freebsd-src.git
   cd freebsd-src
   make -j8 buildkernel DEBUG_FLAGS-g

   # This installs the kernel under another name, `kernel.drm`. Thus,
you keep the default kernel
   # in case of trouble.
   sudo make installkernel DEBUG_FLAGS=-g INSTKERNNAME=kernel.drm

   2.

   Should be


You need to checkout the FreeBSD src branch I mentionned,
linuxkpi-changes-for-drm-from-5.11, and compile a kernel from that branch:

git clone -b linuxkpi-changes-for-drm-from-5.11
https://github.com/dumbbell/freebsd-src.git
cd freebsd-src
make -j8 buildkernel DEBUG_FLAGS=-g

# This installs the kernel under another name, `kernel.drm`. Thus, you keep
the default kernel
# in case of trouble.
sudo make installkernel DEBUG_FLAGS=-g INSTKERNNAME=kernel.drm

**Comment by nevillehay** [comment]
There should be an = after DEBUG_FLAGS

**Comment by dumbbell** [comment]
Thank you! I just fixed the comment.

**Comment by nevillehay** [comment]
I saw one message

Driver bug unable to set devclass

**Comment by nevillehay** [comment]
But I noticed in sndstat I have 2 HDMI devices

**Comment by dumbbell** [comment]
> I saw one message
> 
> Driver bug unable to set devclass

Were you trying to kldload the driver? Could you please share the output of dmesg(8) after doing that?

**Comment by dumbbell** [comment]
Just a general comment: after you try to compile the code, kldload the driver or start X.Org or a Wayland compositor and want to report feedback or ask for help, could you please always provide the output of dmesg(8) (also available in `/var/log/messages`), plus the X.Org log file or the output from the Wayland compositor (depending on what applies to you)?

If you get a kernel panic, I'm interested in the most recent `/var/crash/core.txt.*`. If no a core dump is written to disk, then a photo of the kernel stacktrace on screen would be nice as well.

Thank you :-)

**Comment by nevillehay** [comment]
Okay. Will do. By the way there's no drm-xxxx-kmod package needed?

**Comment by nevillehay** [comment]
No worries. You'll have them tonight.

**Comment by nevillehay** [comment]
Here are some files:
[xorg-amdgpuload.txt](https://github.com/freebsd/drm-kmod/files/8976509/xorg-amdgpuload.txt)
This is the xorg log file with amdgpu in rc.conf
This is an xorg log file without amdgpu in rc.conf
[xorg-noamdgpu.txt](https://github.com/freebsd/drm-kmod/files/8976529/xorg-noamdgpu.txt)

Here is var/log/messages
[messages1.txt](https://github.com/freebsd/drm-kmod/files/8976533/messages1.txt)

dmesg with amdgpu in rc.conf
[dmesg1.txt](https://github.com/freebsd/drm-kmod/files/8976544/dmesg1.txt)

I'll send another dmesg after kldload amdgpu

**Comment by nevillehay** [comment]
Here are some files:
[xorg-amdgpuload.txt](https://github.com/freebsd/drm-kmod/files/8976509/xorg-amdgpuload.txt)
This is the xorg log

**Comment by nevillehay** [comment]
dmesg file after loading amdgpu
[dmesg-kldload-amdgpu.txt](https://github.com/freebsd/drm-kmod/files/8976669/dmesg-kldload-amdgpu.txt)

**Comment by nevillehay** [comment]
I think you're correct
Here is pciconf -lv
[pciconf.txt](https://github.com/freebsd/drm-kmod/files/8977022/pciconf.txt)

And pciconf -lv | grep -B 3 -A 2 display
[pciconf-display.txt](https://github.com/freebsd/drm-kmod/files/8977024/pciconf-display.txt)

**Comment by dumbbell** [comment]
It is supported by Linux. The card's PCI ID was added to the driver in the following commit:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=a2e9b1666ea728aac7107e28d2106fce7e2aba5e

It's part of Linux 5.15.

**Comment by andreaf74** [comment]
> It is supported by Linux. The card's PCI ID was added to the driver in the following commit: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=a2e9b1666ea728aac7107e28d2106fce7e2aba5e
> 
> It's part of Linux 5.15.

Too bad :( , I tried your patch for the same reason as @nevillehay  and I got the same results, so I believe we should wait for linuxkpi to be aligned at least to 5.15.

If needed I'll be more than willing to test future upgrades.

Thanks

**Comment by dumbbell** [comment]
Thank you @andreaf74 for taking the time to test!

**Comment by nevillehay** [comment]
Thanks for your help everyone!

**Comment by JonathanxD** [comment]
First, thanks a lot for your hard work, I've been using FreeBSD in “terminal-only mode” for a couple of months, but couldn't do everything I needed to. This week I found this PR and I was able to clone the fork repos, compile and load the module.

> When I suggesting to link, I had a symlink in mind, so something like ln -s /path/to/src /usr/src (assuming /usr/src does not exist). But you don't need both, the exported variable will work just fine.

It didn't worked for me, I had to make a symlink, otherwise the compiler would complain about declaration conflicts under x86_64 headers, looks like it was importing both the system one and the cloned one, but I'm not sure, because clang was reporting the path as `./x86_64` (relative to CWD) instead of something like `/usr/src/`, and the conflicting one was being reported on the cloned source repo.

I don't know if this is because I'm on FreeBSD 13.1-RELEASE, but shouldn't since I specified the source path, probably there still another env that I should've changed.

Apart from this, everything seems to be working, sometimes I get hard kernel crashes on the first boot, but in the second one it just works, and on “logout” as well. 

Xorg seems to not be working as it should, KDE only starts using `startplasma-x11` when I boot using UEFI, but don't in BIOS mode, also, Xorg doesn't detect radeonsi driver correctly and seems to be falling back to llvmpipe, but I'm not sure if this is real, since without this module I couldn't get any graphical interface working even with llvmpipe, it was only possible with [UEFI GOP](https://wiki.osdev.org/GOP) and [BIOS VESA](https://wiki.osdev.org/VESA). But under Wayland, the experience is just as good as in Linux, Sway loads correctly and `glxinfo` reports my graphics card. Even Firefox shows the right information.

swappy-20220728_222221

I'm very excited in seeing this reaching a mature state, and it already looks like it's almost there.

**Comment by dumbbell** [comment]
Thank you very much for testing :-)

The only crash I get is when I exit from Sway so far. Perhaps we hit the same issue. Unfortunately, I don't get a kernel core dump. Do you have core dumps for the panics you have?

I spent quite some time trying to configure Intel AMT to have a remote serial console over the network, but I couldn't get it to work on FreeBSD (it works on Windows on the same laptop).

About X.Org vs. Wayland, I still didn't test X.Org yet.

Beside fixing the panics, the last big piece of work is the I2C. The drivers in this branch currently relies on the GPLv2 implementation. I need to add the missing bits to the implementation in FreeBSD base. That's why I can't rebase this branch on top of the latest `master` because the GPLv2 copy was removed.

**Comment by JonathanxD** [comment]
> Do you have core dumps for the panics you have?

I do, I can get them anytime, just need to keep restarting until they happen, I don't know if there is a way to get those messages in Plain Text after the fact (I didn't configured a swap partition so I can't use regular kernel debugging techniques, and I don't have that much knowledge on FreeBSD).

I'm attaching the dump as an image, but if you know if there is a way to get those as text, it would be way easier to read.

crash

**Comment by sescu54** [comment]
Any ETA on when this will be ready and pushed to the main repository? :)

**Comment by dumbbell** [comment]
> Any ETA on when this will be ready and pushed to the main repository? :)

Hi @sescu54!

No ETA :) I've been quite busy with other things this summer.

What's left to do:
* Fix a panic, perhaps in the i915 driver, which occurs when another panic is being treated. Probably a difference of behavior between the Linux and FreeBSD consoles. So far, I couldn't find the reason.
* Fix a panic in the amdgpu driver triggered when I exit Sway (but it doesn't happen 100% of the time). I didn't look at it yet and it happens rarely now.
* Test with X.Org. Not started yet.

I won't test suspend & resume because it never worked for me, so I can't see regressions :)

Did you test the driver on your hardware?

**Comment by evadot** [comment]
> > Any ETA on when this will be ready and pushed to the main repository? :)
> 
> Hi @sescu54!
> 
> No ETA :) I've been quite busy with other things this summer.
> 
> What's left to do:
> 
>     * Fix a panic, perhaps in the i915 driver, which occurs when another panic is being treated. Probably a difference of behavior between the Linux and FreeBSD consoles. So far, I couldn't find the reason.
> 
>     * Fix a panic in the amdgpu driver triggered when I exit Sway (but it doesn't happen 100% of the time). I didn't look at it yet and it happens rarely now.
> 
>     * Test with X.Org. Not started yet.
> 
> 
> I won't test suspend & resume because it never worked for me, so I can't see regressions :)

 Suspend/Resume works on all my hardware on main and 13.1, please try again.

> Did you test the driver on your hardware?

**Comment by Skyliegirl33** [comment]
I have tested this branch on my machine (RX 6600), and while I'm not sure the GPU driver is at fault, one game in particular refuses to launch and the error is simply "DirectX error". Can't find anything in Wine or mesa that seems to be failing. Issue here; https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=264973

**Comment by dumbbell** [comment]
@evadot said:
> Suspend/Resume works on all my hardware on main and 13.1, please try again.

I just tested again:
1. From the console, with the `i915kms` DRM driver loaded
2. From the console, with no DRM driver loaded and `debug.acpi.suspend_bounce=1`

In the first case, the screen turns off and a few seconds later, the fans start to spin (but not at their max speed). In the second case, the screen remains on and the fans start too. In both cases, the system doesn't make more progress and isn't responsive anymore.

The laptop is a Dell Precision 5540.

**Comment by evadot** [comment]
> @evadot said:
> 
> > Suspend/Resume works on all my hardware on main and 13.1, please try again.
> 
> I just tested again:
> 
>     1. From the console, with the `i915kms` DRM driver loaded
> 
>     2. From the console, with no DRM driver loaded and `debug.acpi.suspend_bounce=1`
> 
> 
> In the first case, the screen turns off and a few seconds later, the fans start to spin (but not at their max speed). In the second case, the screen remains on and the fans start too. In both cases, the system doesn't make more progress and isn't responsive anymore.
> 
> The laptop is a Dell Precision 5540.

Do you have a discrete GPU ?

**Comment by Skyliegirl33** [comment]
> If you didn't try this branch, what driver do you use?

@dumbbell:
Oh, I ported it to work on 13.1. You can find the repos on my GitHub. I did try this one at first, I should mention, so it's not something with mine.

**Comment by dumbbell** [comment]
> > @evadot said:
> > > Suspend/Resume works on all my hardware on main and 13.1, please try again.
> > 
> > 
> > I just tested again:
> > ```
> > 1. From the console, with the `i915kms` DRM driver loaded
> > 
> > 2. From the console, with no DRM driver loaded and `debug.acpi.suspend_bounce=1`
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > In the first case, the screen turns off and a few seconds later, the fans start to spin (but not at their max speed). In the second case, the screen remains on and the fans start too. In both cases, the system doesn't make more progress and isn't responsive anymore.
> > The laptop is a Dell Precision 5540.
> 
> Do you have a discrete GPU ?

Yes, an NVIDIA one in the laptop which I never used. The NVIDIA drivers are not installed.

I also have an AMD GPU in an eGPU external case. It was not connected as part of the test.

**Comment by dumbbell** [comment]
> **Update:** Now that I think of it, I use `acpi_call` to turn the NVIDIA GPU off so it doesn't consume power from the battery. I will try again without this.

@evadot: After leaving the NVIDIA dGPU alone (powered on but not used), I could suspend/resume the laptop with this branch! I didn't test with DRM 5.10.

I also tested a quick X.Org session and it looks to work (X.Org and KDE start). I need to configure everything to test it further.

**Comment by evadot** [comment]
> 

Yeah dGPU was clearly the problem :)
It should work with 5.10 too.

**Comment by dumbbell** [comment]
I'm marking this pull request as ready to review.

There are two panics I see from time to time:
* One in the `amdgpu` driver which I can't explicitly reproduce.
* One in the `i915kms` driver which occurs when the kernel panics elsewhere (i915kms manages vt(4) in my case).

The first one may be repaired in a future version of the `amdgpu` driver as Linux 5.12 is the recommended version for AMD Radeon RX 6xxx GPUs.

The second one is probably because of a difference between the Linux console driver and our vt(4) behavior. I will continue to look into it, but I don't consider it a show stopper for wider testing because it only happens as part of another crash.

File: linuxkpi/bsd/include/linux/mmzone.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+#ifndef _LINUX_MMZONE_H
+#define _LINUX_MMZONE_H
+
+#include <linux/mm_types.h>
+#include <linux/page-flags.h>
+
+#define MAX_ORDER	11
+
+#endif
```
Comment (evadot) [inline linuxkpi/bsd/include/linux/mmzone.h:11]: This should be in base.

File: linuxkpi/gplv2/include/linux/dma-buf-map.h [hunk 2]
```diff
@@ -0,0 +1,266 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
```
Comment (evadot) [inline linuxkpi/gplv2/include/linux/dma-buf-map.h:1]: No new GPLv2 files please.

**Review by evadot — CHANGES_REQUESTED** [review]

File: linuxkpi/bsd/include/linux/mmzone.h (hunk 1 above)
Comment (dumbbell) [reply linuxkpi/bsd/include/linux/mmzone.h:11]: Ok

File: linuxkpi/gplv2/include/linux/dma-buf-map.h (hunk 2 above)
Comment (dumbbell) [reply linuxkpi/gplv2/include/linux/dma-buf-map.h:1]: Is there a good method to rewrite such a header?

Comment (dumbbell) [reply linuxkpi/gplv2/include/linux/dma-buf-map.h:1]: OpenBSD provides a public domain implementation, I will take this one.

File: linuxkpi/bsd/include/linux/mmzone.h (hunk 1 above)
Comment (dumbbell) [reply linuxkpi/bsd/include/linux/mmzone.h]: Done.

File: linuxkpi/gplv2/include/linux/dma-buf-map.h (hunk 2 above)
Comment (dumbbell) [reply linuxkpi/gplv2/include/linux/dma-buf-map.h]: Done.

**Comment by dumbbell** [comment]
I submitted pull request #207 to fix build again FreeBSD 14-CURRENT 1400071 (there was a change w.r.t. sysctls).

This branch is now rebased on top of that pull request.

**Comment by evadot** [comment]
I'll test this branch early next week on the hardware that I have.
Is the branch bisectable ? (i.e. if there was some FreeBSD specific code that break compilation was it commited in the same linux patch etc ... ?)

**Comment by dumbbell** [comment]
I will double check tomorrow. What do you prefer? I suppose, having the build fix in the original Linux commit. I can probably split or squash relevant commits.

**Comment by evadot** [comment]
> I will double check tomorrow. What do you prefer? I suppose, having the build fix in the original Linux commit. I can probably split or squash relevant commits.

Squashing the fix with the Linux commit is easier and what I did in the past.
I just don't fix if the problem was in Linux but I usually directly apply the fix commit just after (sometimes it's way after).

**Comment by dumbbell** [comment]
> Squashing the fix with the Linux commit is easier and what I did in the past.

I squashed the few build fix commits with into the commits they were fixing, and rebased onto the latest `master`.

**Comment by evadot** [comment]
You didn't opened reviews for base linuxkpi ?

**Comment by dumbbell** [comment]
No, and neither for firmware updates. Should I submit something to Phabricator? Are GitHub PR ok too? I didn't follow if the latter is supported or not.

**Comment by evadot** [comment]
> No, and neither for firmware updates. Should I submit something to Phabricator? Are GitHub PR ok too? I didn't follow if the latter is supported or not.

src commit should be done on reviews.freebsd.org for the firmware open a PR on github.com/freebsd/drm-kmod-firmware

**Comment by dumbbell** [comment]
Here is the review for the changes to linuxkpi:
https://reviews.freebsd.org/D36869

**Comment by dumbbell** [comment]
Here is the firmware update pull request:
freebsd/drm-kmod-firmware#25

**Comment by nevillehay** [comment]
Everything compiled ok but it's not working. The module doesn't seem to be
loading properly. My card is Navi 24. There's support for this card? I'll
do a reinstall and try again. I do kldload amdgpu but nothing happens.

**Comment by dumbbell** [comment]
@nevillehay: Could you pelase share the entry of your GPU in the output of `pciconf -lv`?

Here is an example with my GPU:
```
...
vgapci2@pci0:9:0:0:	class=0x030000 rev=0xc1 hdr=0x00 vendor=0x1002 device=0x73df subvendor=0x1da2 subdevice=0xe445
    vendor     = 'Advanced Micro Devices, Inc. [AMD/ATI]'
    device     = 'Navi 22 [Radeon RX 6700/6700 XT/6750 XT / 6800M]'
    class      = display
    subclass   = VGA
...
```

You can look for your entry by searching for things like "AMD" or "display".

**Comment by dumbbell** [comment]
> I'm using the RX 6500 XT now on scfb.

I think this one was added to Linux 5.12.

**Comment by nevillehay** [comment]
> @nevillehay: Could you pelase share the entry of your GPU in the output of `pciconf -lv`?
> 
> Here is an example with my GPU:
> 
> ```
> ...
> vgapci2@pci0:9:0:0:	class=0x030000 rev=0xc1 hdr=0x00 vendor=0x1002 device=0x73df subvendor=0x1da2 subdevice=0xe445
>     vendor     = 'Advanced Micro Devices, Inc. [AMD/ATI]'
>     device     = 'Navi 22 [Radeon RX 6700/6700 XT/6750 XT / 6800M]'
>     class      = display
>     subclass   = VGA
> ...
> ```
> 
> You can look for your entry by searching for things like "AMD" or "display".

OK. Can you please wait? I have to put the card in, I'll do it asap. Thanks for your work.

**Comment by nevillehay** [comment]
> > @nevillehay: Could you pelase share the entry of your GPU in the output of `pciconf -lv`?
> > Here is an example with my GPU:
> > ```
> > ...
> > vgapci2@pci0:9:0:0:	class=0x030000 rev=0xc1 hdr=0x00 vendor=0x1002 device=0x73df subvendor=0x1da2 subdevice=0xe445
> >     vendor     = 'Advanced Micro Devices, Inc. [AMD/ATI]'
> >     device     = 'Navi 22 [Radeon RX 6700/6700 XT/6750 XT / 6800M]'
> >     class      = display
> >     subclass   = VGA
> > ...
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > You can look for your entry by searching for things like "AMD" or "display".
> 
[pciconf.txt](https://github.com/freebsd/drm-kmod/files/9846340/pciconf.txt)

> OK. Can you please wait? I have to put the card in, I'll do it asap. Thanks for your work.

**Comment by nevillehay** [comment]
> > > @nevillehay: Could you pelase share the entry of your GPU in the output of `pciconf -lv`?
> > > Here is an example with my GPU:
> > > ```
> > > ...
> > > vgapci2@pci0:9:0:0:	class=0x030000 rev=0xc1 hdr=0x00 vendor=0x1002 device=0x73df subvendor=0x1da2 subdevice=0xe445
> > >     vendor     = 'Advanced Micro Devices, Inc. [AMD/ATI]'
> > >     device     = 'Navi 22 [Radeon RX 6700/6700 XT/6750 XT / 6800M]'
> > >     class      = display
> > >     subclass   = VGA
> > > ...
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > You can look for your entry by searching for things like "AMD" or "display".
> 
> [pciconf.txt](https://github.com/freebsd/drm-kmod/files/9846340/pciconf.txt)
> 
> > OK. Can you please wait? I have to put the card in, I'll do it asap. Thanks for your work.
[pciconf.txt](https://github.com/freebsd/drm-kmod/files/9846341/pciconf.txt)

**Comment by nevillehay** [comment]
The RX 6600 (Navi 23) is supported, correct?

**Comment by dumbbell** [comment]
> The RX 6600 (Navi 23) is supported, correct?

Correct. I believe it was added to Linux 5.12. See #208 which is nearly complete.

**Comment by evadot** [comment]
Build ok, can you rebase your branch so I can merge this ?
Thanks,

**Comment by dumbbell** [comment]
The branch is already rebased on the latest `master`. Did you mean something else?

**Review by evadot — APPROVED** [review]

**Comment by evadot** [comment]
> The branch is already rebased on the latest `master`. Did you mean something else?

Looks like it's just github that can't handle big merge request, I'll do this by hand later today or this weekend :)

**Comment by patmaddox** [comment]
> 1. You need to checkout the FreeBSD src branch I mentionned, [`linuxkpi-changes-for-drm-from-5.11`](https://github.com/dumbbell/freebsd-src/tree/linuxkpi-changes-for-drm-from-5.11)

@dumbbell that branch no longer exists - has it been merged to freebsd-src?

**Comment by dumbbell** [comment]
@patmaddox: Yes, all required changes to linuxkpi for Linux 5.11 were committed to `main`. There are still linuxkpi patches I need to work on for Linux 5.12. See #208.

**Pull request merged**